src/Sylius/Component/Inventory/Model/StockItem.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Inventory\Model;
+
+use Doctrine\Common\Collections\ArrayCollection;
+
+/**
+ * Stock item model.
+ *
+ * @author Paweł Jędrzejewski <pawel@sylius.org>
+ */
+class StockItem implements StockItemInterface
+{
+    /**
+     * Stock item id.
+     *
+     * @var mixed
+     */
+    protected $id;
+
+    /**
+     * Stockable object.
+     *
+     * @var StockableInterface
+     */
+    protected $stockable;
+
+    /**
+     * Stock location.
+     *
+     * @var StockLocationInterface
+     */
+    protected $location;
+
+    /**
+     * On hand quantity.
+     *
+     * @var integer
+     */
+    protected $onHand = 0;
+
+    /**
+     * On hold quantity.
+     *
+     * @var integer
+     */
+    protected $onHold = 0;
+
+    /**
+     * Stock movements.
+     *
+     * @var StockMovementInterface[]
+     */
+    protected $stockMovements;
+
+    /**
+     * Available on demand?
+     *
+     * @var Boolean
+     */
+    protected $availableOnDemand = false;
+
+    /**
+     * Creation time.
+     *
+     * @var \DateTime
+     */
+    protected $createdAt;
+
+    /**
+     * Last update time.
+     *
+     * @var \DateTime
+     */
+    protected $updatedAt;
+
+    /**
+     * Constructor.
+     */
+    public function __construct()
+    {
+        $this->stockMovements = new ArrayCollection();
+        $this->createdAt = new \DateTime();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getStockable()
+    {
+        return $this->stockable;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setStockable(StockableInterface $stockable)
+    {
+        $this->stockable = $stockable;
+
+        return $this;
```

Comment:
As this it's done in everywhere other touched places, cause ["fluent interfaces are evil"](http://ocramius.github.io/blog/fluent-interfaces-are-evil/) :wink: